src/openrct2-ui/windows/StaffList.cpp

```diff
@@ -334,19 +334,20 @@ void window_staff_list_update(rct_window* w)
  *
  *  rct2: 0x006BD990
  */
-static void window_staff_list_tooldown(rct_window* w, rct_widgetindex widgetIndex, int32_t x, int32_t y)
+static void window_staff_list_tooldown(rct_window* w, rct_widgetindex widgetIndex, ScreenCoordsXY screenCoords)
 {
     if (widgetIndex == WIDX_STAFF_LIST_SHOW_PATROL_AREA_BUTTON)
     {
         int32_t selectedPeepType = _windowStaffListSelectedTab;
 
         int32_t direction;
         TileElement* tileElement;
-        footpath_get_coordinates_from_pos(x, y, &x, &y, &direction, &tileElement);
-        if (x == LOCATION_NULL)
+        footpath_get_coordinates_from_pos(
+            screenCoords.x, screenCoords.y, &screenCoords.x, &screenCoords.y, &direction, &tileElement);
```

Comment:
Create a new local variable for the return values (the &screenCoords) as the return value may not be in `ScreenCoordsXY`

Reply:
Done

Reply:
Not too sure if this will compile because although the value should be in CoordsXY range I suspect the function expects a int16_t pointer not a int32_t.

Reply:
It did compile incrementally and from clean, but it should, as it expects `int32_t*`